### PR TITLE
Use older function version as template when creating new one

### PR DESCRIPTION
**Pull request by kerkkoheiskanen**
When creating a new version with only the classification as input, 
use the last function version of that classification as a template.

refs #284

### CODE REVIEW

File: metarecord/tests/test_api.py [hunk 1]
```diff
@@ -1852,6 +1853,49 @@ def test_function_post_when_not_allowed(post_function_data, user_api_client):
...
+        response_data_json['phases'][0].pop(attr)
+        function_data['phases'][0].pop(attr)
+    assert response_data_json['phases'][0] == function_data['phases'][0]
+    response_data_json.pop('phases')
+    function_data.pop('phases')
+
+    new_phase = Function.objects.filter(classification=new_classification).latest_version().first().phases.first()
+    assert (new_phase.uuid == phase.uuid)
+    assert new_phase.actions.first().uuid == phase.actions.first().uuid
+    assert new_phase.actions.first().records.first().uuid == phase.actions.first().records.first().uuid
```
Comment (ehaivala) [inline metarecord/tests/test_api.py:1891]: Phases, actions and records should have unique uuid. The uuid non-uniqueness affects only classifications and functions which are versioned.

File: metarecord/views/function.py [hunk 2]
```diff
@@ -78,27 +73,52 @@ def get_fields(self):
...
+                    action.save()
+                    for record in records:
+                        action.records.add(record)
+                    actions.append(action)
+
+                phase.pk = None
+                phase.save()
+                for action in actions:
+                    phase.actions.add(action)
+                function.phases.add(phase)
```
Comment (ehaivala) [inline metarecord/views/function.py:105]: Follow up to the previous comment. Phases, actions and records should be assigned newly generated uuid v4 before saving.

File: metarecord/tests/test_api.py (hunk 1 above)
Comment (kerkkoheiskanen) [reply metarecord/tests/test_api.py:1891]: Ah got it mixed up. Will do.

**Review by tuomas777 — APPROVED** [review]
Looks good as far as I can tell, good job! 👍

**Pull request merged**